django/contrib/gis/geos/coordseq.py

```diff
@@ -157,7 +191,7 @@ def kml(self):
     def tuple(self):
         "Return a tuple version of this coordinate sequence."
         n = self.size
+        get_point = self._point_getter
         if n == 1:
-            return self[0]
-        else:
-            return tuple(self[i] for i in range(n))
+            return get_point(0)
+        return tuple(get_point(i) for i in range(n))
```

Comment:
One thing to check is that `tuple([])` might be fasted than without the `[]` (suggested in https://github.com/django/django/pull/7730#issuecomment-269043053).